pkg/block/block.go

```diff
@@ -42,9 +42,28 @@ const (
 	DebugMetas = "debug/metas"
 )
 
+// downloadDirFn is a function type that a function which downloads a directory from remote object
+// storage must implement. Used for mocking.
+type downloadDirFn func(context.Context, log.Logger, objstore.BucketReader, string, string, []string) error
+
 // Download downloads directory that is mean to be block directory.
-func Download(ctx context.Context, logger log.Logger, bucket objstore.Bucket, id ulid.ULID, dst string) error {
-	if err := objstore.DownloadDir(ctx, logger, bucket, id.String(), dst); err != nil {
+func Download(ctx context.Context, logger log.Logger, bucket objstore.Bucket, m *metadata.Meta, dst string, downloadDir downloadDirFn) error {
+	ignoredPaths := []string{}
+	if len(m.Thanos.Hashes) > 0 {
```

Comment:
No need for this `if`. The next `for` will be a no-op if `m.Thanos.Hashes` is empty.